Issue-3393- Documentation for multistore setups (#8247)
magento/devdocs: src/guides/v2.3/install-gde/install/cli/install-cli-subcommands-maint.md

`--none` clears the list.

## For Multistore setups {#instgde-cli-multistore}

> When we need different layout and localized content, general speculation is that there is not an out of the box feature. But, there is one. The error processor accepts an parameter to change the error page template.

Reviewer: Documents should be factual and directive only. Please rewrite this just to say how this is done and when.

Author: I have changed this now

> ```php

Reviewer: Where does this PHP come from?

Author: There was a lint error that was indicating that I have to specify the type of the language I used here. That's why I had to add that php there

Reviewer: What I meant was, where does the PHP code come from? You haven't explained what file and the context

if (isset($_GET['skin'])) {
    $this->_setSkin($_GET['skin']);
}
```

This can be passed via rewrite rule in .htaccess and needs no further changes.

> ## For $_GET['skin'] parameter {#instgde-cli-multistore-getskin}

Reviewer: This heading has equal weight to the previous. It looks like a completely separate section. if this is part of the previous section then lower the weight so it's clear it is part of the previous section

*  Check if the `.maintenance.flag` exists.
*  Check for the host (or any other form like the ENV variables)
*  Set the parameter.

> ## Rewrite rule in .htaccess {#instgde-cli-multistore-rewrite-rule}

Reviewer: See previous heading

*  RewriteCond `%{DOCUMENT_ROOT}/var/.maintenance.flag -f`
*  RewriteCond `%{HTTP_HOST} ^sub.example.com$`
*  RewriteCond `%{QUERY_STRING} !(^|&)skin=sub(&|$)` [NC]
*  RewriteRule `^ %{REQUEST_URI}?skin=sub` [L]

Then, copy the following files:

*  `pub/errors/default/503.phtml` to `pub/errors/sub/503.phtml`

> *  `pub/errors/default/css/styles.css` to `pub/errors/default/css/styles.css`

Reviewer: Both file names are the same

Author: I have changed the file path of the target file to 'sub' directory. @sreichel Could you please, confirm this?

Reviewer 2: > 
> 
> I have changed the file path of the target file to 'sub' directory. @sreichel Could you please, confirm this?

Sorry for being late ... it should be `pub/errors/sub/...`. Thanks for working on this.

> *  Make your edits in these files.

Reviewer: This is not a file, so please remove from the list and make separate

> Make sure, the Directory name has to match the URL parameter.

Reviewer: it is unclear what this directive is saying - What Directly? What parameter?

{:.bs-callout-info}

> The nginx setting should be added for multistore setups.

Reviewer: What setting?

Author: I have added a hyperlink for the topic now

`--none` clears the list.

## For Multistore setups {#instgde-cli-multistore}

> If we want to setup multiple stores with different layout and localized content for each and every store, we have to use an out of the box feature to achieve it. That is by passing the `$_GET['skin']` parameter to the intended processor. For example, in a sample scenario, The error processor accepts an parameter to change the error page template. Here, we have considered a `503` type error template file for which we have provided the localized content. Below are the general steps to follow,

Reviewer: ```suggestion
If we want to setup multiple stores with a different layout and localized content for each and every store, we have to use an out of the box feature to achieve it. This is done by passing the `$_GET['skin']` parameter to the intended processor. For example, the error processor accepts a parameter to change the error page template. In the example below we are using a `503` type error template file for which we have provided the localized content. The following steps show how this is done:
```

```php
if (isset($_GET['skin'])) {
    $this->_setSkin($_GET['skin']);
}
```

> This can be passed via rewrite rule in .htaccess and needs no further changes.

Reviewer: This sentence seems out of place as you can't put PHP in a .htaccess file, which it infers, please explain what this refers to.

# For $_GET['skin'] parameter {#instgde-cli-multistore-getskin}

*  Check if the `.maintenance.flag` exists.

> *  Check for the host (or any other form like the ENV variables)

Reviewer: The second and third instruction are confusing as it's not clear what's being checked, where it's being checked and what gets set.

*  Set the parameter.

# Rewrite rule in .htaccess {#instgde-cli-multistore-rewrite-rule}

*  RewriteCond `%{DOCUMENT_ROOT}/var/.maintenance.flag -f`
*  RewriteCond `%{HTTP_HOST} ^sub.example.com$`
*  RewriteCond `%{QUERY_STRING} !(^|&)skin=sub(&|$)` [NC]
*  RewriteRule `^ %{REQUEST_URI}?skin=sub` [L]

Then, copy the following files:

*  `pub/errors/default/503.phtml` to `pub/errors/sub/503.phtml`
*  `pub/errors/default/css/styles.css` to `pub/errors/sub/styles.css`

> Make your edits in these files.

Reviewer: What edits?

> Make sure, the Directory name has to match the URL parameter. Here it indicates the directory `sub` which is specified as a parameter in the rewrite rule above (`skin=sub`)

Reviewer: ```suggestion
In the file example above make careful note of the the directory used under the `errors` directory. The directory name has to match the URL parameter as indicated in the RewriteRule.  In the example above the directory `sub` is used which is specified as a parameter in the RewriteRule (`skin=sub`)
```

`--none` clears the list.

## For Multistore setups {#instgde-cli-multistore}

If we want to setup multiple stores with a different layout and localized content for each and every store, we have to use an out of the box feature to achieve it. This is done by passing the `$_GET['skin']` parameter to the intended processor. For example, the error processor accepts a parameter to change the error page template. In the example below we are using a `503` type error template file for which we have provided the localized content. The following steps show how this is done:

> The Constructor of the `Error_Processor` class accepts a `skin` POST parameter to change layout.

Reviewer: ```suggestion
The constructor of the `Error_Processor` class accepts a `skin` POST parameter to change layout.
```

```php
if (isset($_GET['skin'])) {
    $this->_setSkin($_GET['skin']);
}
```

This can be added as rewrite rules in the `.htaccess` file that appends a `skin` parameter to the URL.

# For $_GET['skin'] parameter {#instgde-cli-multistore-getskin}

*  Check if the `.maintenance.flag` exists.

> *  Note out the host address, that refers to the `HTTP_HOST` (or any other form like the ENV variables)

Reviewer: ```suggestion
*  Note the host address, that refers to the `HTTP_HOST` (or any other form like the ENV variables)
```

*  Set the parameter by using the rewrite rules below.

# Rewrite rule in .htaccess {#instgde-cli-multistore-rewrite-rule}

*  RewriteCond `%{DOCUMENT_ROOT}/var/.maintenance.flag -f`
*  RewriteCond `%{HTTP_HOST} ^sub.example.com$`
*  RewriteCond `%{QUERY_STRING} !(^|&)skin=sub(&|$)` [NC]
*  RewriteRule `^ %{REQUEST_URI}?skin=sub` [L]

Then, copy the following files:

*  `pub/errors/default/503.phtml` to `pub/errors/sub/503.phtml`
*  `pub/errors/default/css/styles.css` to `pub/errors/sub/styles.css`

Make your edits in these files such as providing localized content in the `503.phtml` file and adding custom styles in the `styles.css` file.

> In the file example above make careful note of the the directory used under the `errors` directory. The directory name has to match the URL parameter as indicated in the RewriteRule. In the example above the directory `sub` is used which is specified as a parameter in the RewriteRule (`skin=sub`)

Reviewer: ```suggestion
In the file example above make careful note of the directory used under the `errors` directory. The directory name has to match the URL parameter as indicated in the RewriteRule. In the example above the directory `sub` is used which is specified as a parameter in the RewriteRule (`skin=sub`)
```

`--none` clears the list.

## For Multistore setups {#instgde-cli-multistore}

If you want to setup multiple stores with a different layout and localized content for each store, this is done by passing the `$_GET['skin']` parameter to the intended processor In the example below, we are using a `503` type error template file, which requires localized content.

The constructor of the `Error_Processor` class accepts a `skin` GET parameter to change layout:

```php
if (isset($_GET['skin'])) {
    $this->_setSkin($_GET['skin']);
}
```

This can also be added a rewrite rule in the `.htaccess` file that will append a `skin` parameter to the URL.

### $_GET['skin'] parameter {#instgde-cli-multistore-getskin}

To use the `skin` parameter:

1. Check if the `.maintenance.flag` exists
1. Note the host address, that refers to the `HTTP_HOST`, or any other variable such as ENV variables

> 1. Check if the `skin` parameter exists

Reviewer 3: ```suggestion
1. Check if the `skin` parameter exists.
```

1. Set the parameter by using the rewrite rules below.

> This is what it looks like as a rewrite rule:

Reviewer 3: ```suggestion
See the following for examples of rewrite rules:
```

*  RewriteCond `%{DOCUMENT_ROOT}/var/.maintenance.flag -f`
*  RewriteCond `%{HTTP_HOST} ^sub.example.com$`
*  RewriteCond `%{QUERY_STRING} !(^|&)skin=sub(&|$)` [NC]
*  RewriteRule `^ %{REQUEST_URI}?skin=sub` [L]

Copy the following files:

*  `pub/errors/default/503.phtml` to `pub/errors/sub/503.phtml`
*  `pub/errors/default/css/styles.css` to `pub/errors/sub/styles.css`

Edit these files to provide localized content in the `503.phtml` file and custom styling in the `styles.css` file.

> Ensure your paths point to your `errors` directory. The directory name must match the URL parameter indicated in the RewriteRule. In the example above, the `sub` directory is used, which is specified as a parameter in the RewriteRule (`skin=sub`)

Reviewer 3: ```suggestion
Ensure your paths point to your `errors` directory. The directory name must match the URL parameter indicated in the RewriteRule. In the previous example, the `sub` directory is used, which is specified as a parameter in the RewriteRule (`skin=sub`).
```

{:.bs-callout-info}

> The [nginx]({{ page.baseurl }}/config-guide/multi-site/ms_nginx.html) setting should be added for multistore setups.

Reviewer 3: "should" or "must"? 

Is this simply a recommendation, or is it required? If required, it must be "must" 😉
